Src/VimCore/CommandRunner.fsi

```diff
@@ -8,5 +8,5 @@ open Vim
 
 type internal CommandRunner =
     interface ICommandRunner
-    new: ITextView * IRegisterMap * IMotionCapture * IVimLocalSettings * ICommandUtil * IStatusUtil * VisualKind * KeyRemapMode -> CommandRunner
+    new: IVimBufferData * IMotionCapture * ICommandUtil * VisualKind * KeyRemapMode -> CommandRunner

```

Comment:
I'm guessing there are a lot of small but useful simplifications like this we could make now. Most of these ctors were written before I'd worked out types like `IVimBufferData` and was far into mock based testing.

Reply:
I specifically needed a centralized place to put the `UseVirtualSpace` property and it went in `VimTextBuffer` which is the lowest level at which we can combine the settings with the mode to answer the question "Should I be using virtual space right now?" And I was sure that `VimTextBuffer` was OK to pass and many similar services use it. However, once I added `VimTextBuffer`, it became clear that other resources were now directly accessible without separate parameters.

I try to avoid gratuitous changes, but adding `VimTextBuffer` without removing `IVimLocalSettings` seemed like leaving the job half-finished!

Reply:
I agree. This is definitely an improvement.